core/kernel/trust_spec.rb

```diff
@@ -24,4 +24,20 @@
       o.trust.should equal(o)
     end
   end
+
+  ruby_version_is "2.7"..."3.0" do
+    it "is a no-op" do
+      o = Object.new.untrust
+      o.trust
+      o.should_not.untrusted?
+    end
```

Comment:
Not sure whether I understand what this case checks.

It might check that `trust` method doesn't change object state. But `o.should_not.untrusted?` assertion checks that `untrusted?` returns `false`. Wouldn't it be expected behavior if `trust` worked and changed an object state?

Reply:
Put on `o.should_not.untrusted?` above the `o.trust` to show nothing happens